src/main/java/org/truffleruby/core/thread/ThreadManager.java

```diff
@@ -118,7 +118,7 @@ public void restartMainThread(Thread mainJavaThread) {
 
     public Thread createJavaThread(Runnable runnable) {
         if (context.getOptions().SINGLE_THREADED) {
-            throw new RaiseException(context.getCoreExceptions().internalError("threads not allowed in single-threaded mode", null));
+            throw new RaiseException(context.getCoreExceptions().internalError("threads not allowed in single-threaded mode", null), true);
```

Comment:
This might make sense as a SecurityError.

Reply:
Fixing.